batch/batch/worker/worker.py

```diff
@@ -514,38 +473,38 @@ async def pull_image(self):
 
         try:
             if not is_google_image:
-                await self.ensure_image_is_pulled()
-            elif is_public_image:
-                auth = await self.batch_worker_access_token()
-                await self.ensure_image_is_pulled(auth=auth)
+                # Explicitly default to dockerhub
+                if self.image_ref.domain is None:
+                    self.image_ref.domain = 'docker.io'
+                    self.image_ref_str = str(self.image_ref)
+                await check_exec_output('podman', 'pull', self.image_ref_str)
             else:
-                # Pull to verify this user has access to this
-                # image.
-                # FIXME improve the performance of this with a
-                # per-user image cache.
-                auth = self.current_user_access_token()
-                await docker_call_retry(MAX_DOCKER_IMAGE_PULL_SECS, f'{self}')(
-                    docker.images.pull, self.image_ref_str, auth=auth
-                )
-        except DockerError as e:
-            if e.status == 404 and 'pull access denied' in e.message:
+                if is_public_image:
+                    auth = await self.batch_worker_access_token()
+                else:
+                    # Pull to verify this user has access to this
+                    # image.
+                    # FIXME improve the performance of this with a
+                    # per-user image cache.
+                    auth = self.current_user_access_token()
+                os.makedirs(self.container_scratch, exist_ok=True)
```

Comment:
I feel like we should do this when the container class is created, not inside the image pull function.